### ISSUE TITLE
Validator: Should check def-use dominance rules for OpPhi (variable,parent) operands

### ISSUE THREAD

**Issue by dneto0** [KhronosGroup/SPIRV-Tools#288]
See example in disabled (proposed) test DISABLED_PhiVariableDefMustComeFromBlockDominatingThePredecessorBad

```
%func        = OpFunction %voidt None %vfunct
%entry       = OpLabel
               OpBranchConditional %false %if_true %if_false

 %if_true     = OpLabel
%true_copy   = OpCopyObject %boolt %false
                OpBranch %exit
 
%if_false    = OpLabel
%false_copy  = OpCopyObject %boolt %false
               OpBranch %exit

; The (variable,Id) pairs are swapped.
%exit        = OpLabel
%value       = OpPhi %boolt %true_copy %if_false %false_copy %if_true
               OpReturn
               OpFunctionEnd
```

We should be able to check that for a particular (variable,parent) pair of an OpPhi, that %variable is defined in a block that dominates %parent.  That check should generate an error for the above case.

**Comment by umar456** [comment]
@dneto0 I think this issue can be closed

**Comment by dneto0** [comment]
Fixed with 489204cf0 (and  677ae7fbe)
